### ISSUE TITLE
Creating an 'experiment' class.

### ISSUE THREAD

**Issue by clabornd** [clabornd/mlflow-examples#2]
First off, I have xgboost under the `sklearn` group, should probably add it to it's own config group.

I noticed xgboost has its own quirks when doing test set evaluation.  Specifically it needs classes to be converted to [0,1,2].  I could keep adding little conditional checks for how to do the evaluation and report metrics, but it is probably better to make a high level 'experiment' class that has its own `evaluate` method.
